### PR TITLE
Provide support for custom neuroglancer URL

### PR DESCRIPTION
**Pull request by aaronkanzer**

### CODE REVIEW

**Comment by kabilar** [comment]
Thank you, @aaronkanzer.  Thats good to know.  I agree that we don't need to prioritize merging this upstream to `google/neuroglancer` now.  Should we merge it to `lincbrain/neuroglancer`?

File: src/util/http_path_completion.ts [hunk 1]
```diff
@@ -183,13 +182,17 @@ export async function completeHttpPath(
...
+    if (protocol === "https" && path.length > 0) {
+      return await getS3PathCompletions(host, path, cancellationToken);
     }
+    // if (protocol === "https" && path.length > 0) {
+    //   return await getHtmlPathCompletions(
+    //     parsedUrl,
+    //     cancellationToken,
+    //     credentialsProvider,
+    //   );
+    // }
```
Comment (kabilar) [inline src/util/http_path_completion.ts:195]: I lean towards removing commented code.

```suggestion
```

File: python/neuroglancer/server.py [hunk 2]
```diff
@@ -199,6 +199,7 @@ def get(self, viewer_token, path):
             query = self.request.query
             if query:
                 query = f"?{query}"
+                # Aaron
```
Comment (kabilar) [inline python/neuroglancer/server.py:202]:
```suggestion
```

File: .idea/.gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
```
Comment (kabilar) [inline .idea/.gitignore:3]:
```suggestion
# Default ignored files
/shelf/
/workspace.xml
.DS_Store
```

File: .DS_Store
Comment (kabilar) [inline .DS_Store:1]: Let's remove this file.

File: Dockerfile.dev [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+COPY . .
+
+# Install project dependencies
+# RUN npm i
+
+# Vue CLI serves on port 8080 by default, expose that port
+# EXPOSE 8080
+
+# Command to run the app using npm
+# CMD ["npm", "run", "dev-server"]
```
Comment (kabilar) [inline Dockerfile.dev:25]: Should we remove these commands?

**Review by kabilar — COMMENTED** [review]
Thanks @aaronkanzer.  I provided a few suggestions, but I don't know enough TypeScript to provide a thorough review.

**Comment by aaronkanzer** [comment]
@kabilar -- thanks for the review -- after re-reading my code here, I think I might hold off on merging this until we have MIT storage onboard (e.g. the way that we fetch assets from Engaging might be drastically different)

Considering our neuroglancer instance is stable, I figured punting this for now would be alright -- let me know if you'd like me to close

**Comment by aaronkanzer** [comment]
@kabilar -- closing since we merged files we wanted in the short-term in https://github.com/lincbrain/neuroglancer/pull/7

File: src/util/http_path_completion.ts [hunk 5]
```diff
@@ -183,13 +182,17 @@ export async function completeHttpPath(
...
       );
     }
-    if ((protocol === "http" || protocol === "https") && path.length > 0) {
-      return await getHtmlPathCompletions(
-        parsedUrl,
-        cancellationToken,
-        credentialsProvider,
-      );
+    // Catch all for CloudFront neuroglancer
+    if (protocol === "https" && path.length > 0) {
```
Comment (aaronkanzer) [inline src/util/http_path_completion.ts:186]: Comment to note code for Linda Hedges et. al

**Pull request closed without merging**